File: app/src/main/java/com/tkdev/weatherapp/common/dialogs/SearchDialog.java
package com.tkdev.weatherapp.common.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.tkdev.weatherapp.R;

import java.util.Objects;

public class SearchDialog extends DialogFragment {

    public interface SearchDialogListener {
        void onAboutDialogPositiveClick(String text);

        void onAboutDialogNegativeClick(String text);
    }

    SearchDialogListener listener;


    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            listener = (SearchDialogListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(getActivity().toString()
                    + " must implement SearchDialogListener");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(Objects.requireNonNull(getActivity()));
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.fragment_search, null);
        final EditText city = dialogView.findViewById(R.id.city_search_text);

        builder.setView(dialogView)
                .setPositiveButton(R.string.apply, (dialog, id) -> {
                    listener.onAboutDialogPositiveClick(
                            cityStringRefubrish(city.getText().toString()));

                })
                .setNegativeButton(R.string.cancel, (dialog, which) -> {
                    setCancelable(true);
                    listener.onAboutDialogNegativeClick(getString(R.string.search_dialog_onCancel));
                });


        return builder.create();
    }

    private String cityStringRefubrish(String city) {
        // avoiding different typing in cities - to always have same call
        return city.substring(0, 1).toUpperCase() +
                city.substring(1).toLowerCase();

    }
}

File: app/src/main/java/com/tkdev/weatherapp/common/dialogs/SearchDialog.kt
package com.tkdev.weatherapp.common.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.tkdev.weatherapp.R
import kotlinx.android.synthetic.main.fragment_search.view.*

class SearchDialog : DialogFragment() {

    lateinit var listener: SearchDialogListener

    interface SearchDialogListener {
        fun onSearchPositiveClick(city: String)
        fun onSearchNegativeClick(message: String)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            listener = context as SearchDialogListener
        } catch (e: ClassCastException) {
            throw ClassCastException((context.toString() +
                    " must implement SearchDialogListener"))
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        return activity?.let {
            val builder: AlertDialog.Builder? = AlertDialog.Builder(it)
            val inflater = requireActivity().layoutInflater
            val dialogLayout = inflater.inflate(R.layout.fragment_search, null)
            val editText: EditText = dialogLayout.city_search_text
            builder?.setView(dialogLayout)

            builder?.setPositiveButton(R.string.apply)
            { _, _ ->
                listener.onSearchPositiveClick(editText.text.toString())
            }
            builder?.setNegativeButton(R.string.cancel)
            { _, _ ->
                listener.onSearchNegativeClick(getString(R.string.search_dialog_onCancel))
            }

            builder?.create()
        } ?: throw IllegalStateException("Activity cannot be null")
    }
}